ibm/resource_ibm_is_subnet_network_acl_attachment.go

```diff
@@ -0,0 +1,439 @@
+					if rulex.Code != nil && rulex.Type != nil {
+						icmp[0] = map[string]int{
+							isNetworkACLRuleICMPCode: int(*rulex.Code),
+							isNetworkACLRuleICMPType: int(*rulex.Code),
+						}
+					}
+					rule[isNetworkACLRuleICMP] = icmp
+				}
+			case "*vpcv1.NetworkACLRuleItemNetworkACLRuleProtocolTcpudp":
+				{
+					rulex := rulex.(*vpcv1.NetworkACLRuleItemNetworkACLRuleProtocolTcpudp)
+					rule[isNetworkACLRuleID] = *rulex.ID
+					rule[isNetworkACLRuleName] = *rulex.Name
+					rule[isNetworkACLRuleAction] = *rulex.Action
+					rule[isNetworkACLRuleIPVersion] = *rulex.IPVersion
+					rule[isNetworkACLRuleSource] = *rulex.Source
+					rule[isNetworkACLRuleDestination] = *rulex.Destination
+					rule[isNetworkACLRuleDirection] = *rulex.Direction
+					if *rulex.Protocol == "tcp" {
+						rule[isNetworkACLRuleICMP] = make([]map[string]int, 0, 0)
+						rule[isNetworkACLRuleUDP] = make([]map[string]int, 0, 0)
+						tcp := make([]map[string]int, 1, 1)
+						tcp[0] = map[string]int{
+							isNetworkACLRuleSourcePortMax: checkNetworkACLNil(rulex.SourcePortMax),
+							isNetworkACLRuleSourcePortMin: checkNetworkACLNil(rulex.SourcePortMin),
+						}
+						tcp[0][isNetworkACLRulePortMax] = checkNetworkACLNil(rulex.DestinationPortMax)
+						tcp[0][isNetworkACLRulePortMin] = checkNetworkACLNil(rulex.DestinationPortMin)
+						rule[isNetworkACLRuleTCP] = tcp
+					} else if *rulex.Protocol == "udp" {
+						rule[isNetworkACLRuleICMP] = make([]map[string]int, 0, 0)
+						rule[isNetworkACLRuleTCP] = make([]map[string]int, 0, 0)
+						udp := make([]map[string]int, 1, 1)
+						udp[0] = map[string]int{
+							isNetworkACLRuleSourcePortMax: checkNetworkACLNil(rulex.SourcePortMax),
+							isNetworkACLRuleSourcePortMin: checkNetworkACLNil(rulex.SourcePortMin),
+						}
+						udp[0][isNetworkACLRulePortMax] = checkNetworkACLNil(rulex.DestinationPortMax)
+						udp[0][isNetworkACLRulePortMin] = checkNetworkACLNil(rulex.DestinationPortMin)
+						rule[isNetworkACLRuleUDP] = udp
+					}
+				}
+			case "*vpcv1.NetworkACLRuleItemNetworkACLRuleProtocolAll":
+				{
+					rulex := rulex.(*vpcv1.NetworkACLRuleItemNetworkACLRuleProtocolAll)
+					rule[isNetworkACLRuleID] = *rulex.ID
+					rule[isNetworkACLRuleName] = *rulex.Name
+					rule[isNetworkACLRuleAction] = *rulex.Action
+					rule[isNetworkACLRuleIPVersion] = *rulex.IPVersion
+					rule[isNetworkACLRuleSource] = *rulex.Source
+					rule[isNetworkACLRuleDestination] = *rulex.Destination
+					rule[isNetworkACLRuleDirection] = *rulex.Direction
+					rule[isNetworkACLRuleICMP] = make([]map[string]int, 0, 0)
+					rule[isNetworkACLRuleTCP] = make([]map[string]int, 0, 0)
+					rule[isNetworkACLRuleUDP] = make([]map[string]int, 0, 0)
+				}
+			}
+			rules = append(rules, rule)
+		}
+	}
+	d.Set(isNetworkACLRules, rules)
+	return nil
+}
+
+func resourceIBMISSubnetNetworkACLAttachmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	hasChanged := false
+	if d.HasChange(isNetworkACLID) {
```

Comment:
No need of hasChanged variables since we have only one argument
```
d.HasChange(isNetworkACLID){
subnet := d.Get(isSubnetID).(string)
		networkACL := d.Get(isNetworkACLID).(string)

		// Construct an instance of the NetworkACLIdentityByID model
		networkACLIdentityModel := new(vpcv1.NetworkACLIdentityByID)
		networkACLIdentityModel.ID = &networkACL

		// Construct an instance of the ReplaceSubnetNetworkACLOptions model
		replaceSubnetNetworkACLOptionsModel := new(vpcv1.ReplaceSubnetNetworkACLOptions)
		replaceSubnetNetworkACLOptionsModel.ID = &subnet
		replaceSubnetNetworkACLOptionsModel.NetworkACLIdentity = networkACLIdentityModel
		resultACL, response, err := sess.ReplaceSubnetNetworkACL(replaceSubnetNetworkACLOptionsModel)

		if err != nil {
			log.Printf("[DEBUG] Error while attaching a network ACL to a subnet %s\n%s", err, response)
			return fmt.Errorf("Error while attaching a network ACL to a subnet %s\n%s", err, response)
		}
		log.Printf("[INFO] Updated subnet %s with Network ACL : %s", subnet, *resultACL.ID)
}
return resourceIBMISSubnetNetworkACLAttachmentRead(d, meta)
}
```

Reply:
Corrected this change.